### ISSUE TITLE
Returning back from "Document Info" causes a crash.

### ISSUE THREAD

**Issue by tigran123** [koreader/kindlepdfviewer#253]
Run "reader.lua ." and then select any file, then press "Right" to show "Document Information", then press "Back". The program will terminate with the following error message on the terminal:

```
lua config error: ./fileinfo.lua:139: attempt to call method 'getByKeydef' (a nil value)
```

Tested in the emulator on the current master tree plus the change infont -> infofont in dialog.lua and tfont->tface in HelpPage:show().

**Comment by dpavlin** [comment]
It seems that `FileInfo:init` never gets called, allthough it doesn't make sense... Init gets called, but `self:addAllCommands()` never gets invoked allthouh `self.commands` is `{}` and `not {}` if false and not true.

```
    if not self.commands then
        self:addAllCommands()
   end
```

**Comment by dpavlin** [comment]
Replacing if with `if #self.commands == 0 then` fixes this problem and this seem like lua way to check if table is empty, but I'm also lua newbie, so I'm not completely sure. @houqp is our local lua expert, I'm afraid :-)

**Comment by tigran123** [comment]
Aha, I encountered the difference between "if not table" and "if #table == 0" in my multilevel TOC code already. The first happens after "table=nil", i.e. when table does not exist, while the second after "table = {}", i.e. when it exists as an object, but contains zero initialized elements.

**Comment by houqp** [comment]
@tigran123 is right :) 

An alternative fix can be change following:

``` lua
FileInfo = {
    ...

    commands = {},

    ...
}
```

to:

``` lua
FileInfo = {
    ...

    commands = nil,

    ...
}
```

Since the empty table is useless, why not just set it to nil ;P

**Comment by houqp** [comment]
As for @dpavlin 's question ,  Indeed, `#tablename` is should be used with care on table empty check. It will ignore entries whose key is not a number:

```
> a = {1,2}
> print(#a)
2
> a = {item1=1, item2=2}
> print(#a)
0
```

**Comment by houqp** [comment]
BTW, this bug also supports my urge on developing on the new UI branch. ;P

In the new ui code base, you don't need to copy commands related code in every widget, all the key bindings and key events stuffs are handled (or hided) by the  UI module. Thus avoiding bugs of these kind. So when you are writing a widget, you only need to focus on high level design.

**Comment by ghost** [comment]
I've looked at my not crashing version of fileinfo.lua and found out that  self.commands are defined in the FileInfo-header as nil. So I made one more unnecessary mistake (like missed parenthesis) by manual searching & editing the code via http-interface. Next time I'll just remove the whole text and replace it with the new one. I'm so sorry.

**Comment by houqp** [comment]
Yes, remove the whole text is a better way to add your changes if you use the web interface. However, doing it with Git command line or GUI tools can save you a lot of efforts and time. Anyway, hope you a good holiday :)

**Comment by dpavlin** [comment]
This one is fixed, we can move to #259 now :-)
